### ISSUE TITLE
Can't build just a single tag? 

### ISSUE THREAD

**Issue by peekxc** [airspeed-velocity/asv#1449]
I have a package that takes a long time (relatively) to build. I just want to quickly run a benchmark on tagged commits. I tried doing just one tag for now: 

```bash
asv run v0.5.6 --verbose --no-pull
```

I only want to test one Python version and one commit. No matrix builds. Already, though, this has been running for 10+ minutes just rebuilding the package several times over and over...

Thus I have two bugs to report:  
- Running `asv` on a single tag [per the docs](https://asv.readthedocs.io/en/latest/using.html#benchmarking) doesn't work; it proceeds building the commit 4 times redundantly!
- Environments are not cached or re-used [per the docs](https://asv.readthedocs.io/en/latest/using.html#environments); they just keep getting re-created from scratch every time. 

And one suggestion; after all the builds finally finish, the only message I get is: 

```
   Successfully built scikit-primate
   ...
·· Installing 6f48a75d <main> into virtualenv-py3.11
·· Failed to build the project and import the benchmark suite.
```

This doesn't really help me figure out what's happening in terms of an error. The project did build, so why did the import fail? No traceback, even with verbose? 

Here's my very simple `conf.json`:

```json
{
	"version": 1,
	"project": "primate",
	"project_url": "https://peekxc.github.io/primate/",
	"repo": "https://github.com/peekxc/primate.git",
	"branches": ["main"], 
	"dvcs": "git",
	"environment_type": "virtualenv",
	"pythons": ["3.11"],
	"build_cache_size": 1
}
```

**Comment by HaoZeke** [comment]
OTOH you can specify the commit corresponding to the tag in a file and use that, though the documented variant should be working, that's a bug.

Environment caching should be taking place, but also, perhaps it would be best to use mamba or even the rattler backend (not merged yet) since those environments handle caches better.

I'll have a more complete response in a bit.

**Comment by peekxc** [comment]
Hey @HaoZeke, thanks for replying. 

Yeah I ended up trying a couple things, including actually just adding the commit hashes to a file and benchmarking only those. But even here it does multiple builds, only to eventually report a fail (for unknown reasons, despite the build completing). Then, `asv` proceeds to try building other commits (also for unknown reasons). 

After a couple hours I gave up and decided it would be easier to just run my own benchmarks over time, `asv` seems to be too difficult to configure for my use-case
